server/app.js

```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const path = require('path');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+require('env2')('config.env');
+const reportRoutes = require('./routes/reportRoutes');
+
+const app = express();
+
+const port = process.env.PORT || 5000;
+const host = process.env.HOST || 'localhost';
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+mongoose.connect(process.env.MONGO_URL);
+
+mongoose.connection.on('error', err => {
+  console.log('Mongoose default connection error: ' + err);
+});
+
+mongoose.connection.on('connected', () => {
+  console.log('Mongoose default connection open');
+  app.listen(port, () => {
+    console.log(`App listening on http://${host}:${port}`);
+  });
+});
+
+process.on('unhandledRejection', up => {
```

Comment:
NO lol, unhandled rejection means you did add `.catch()` on a promise which you always should!!!
if the either is about a deprecated promise library then it's something else [this is how](https://github.com/Automattic/mongoose/issues/4951#issuecomment-278081233) you would fix that problem with

```js
mongoose.Promise = require('bluebird')
```

Reply:
Well I have the latest version of node installed on my laptop! it throws warnings if I have errors on my code! It doesn't tell me where the error is unless I add this code here!